### PR TITLE
ffi: add `RoomPinnedEventsChange` enum and a diffing step

### PR DESCRIPTION
**Pull request by jmartinesp**
This is needed to give the clients some context of what happened in the last pinning/unpinning events action so they can display the right info in the timeline events.

- [ ] Public API changes documented in changelogs (optional)

Signed-off-by:

### CODE REVIEW

File: bindings/matrix-sdk-ffi/src/timeline/content.rs [hunk 1]
```diff
@@ -381,7 +394,49 @@ impl From<&matrix_sdk_ui::timeline::AnyOtherFullStateEventContent> for OtherStat
                 };
                 Self::RoomName { name }
             }
-            Content::RoomPinnedEvents(_) => Self::RoomPinnedEvents,
+            Content::RoomPinnedEvents(c) => {
```
Comment (andybalaam) [inline bindings/matrix-sdk-ffi/src/timeline/content.rs:397]: This is a massive chunk, which I'd prefer to have in a separate function, and unit tested, if possible.

**Review by andybalaam — APPROVED** [review]
One suggestion that you can consider optional. Thanks!

File: bindings/matrix-sdk-ffi/src/timeline/content.rs (hunk 1 above)
Comment (jmartinesp) [reply bindings/matrix-sdk-ffi/src/timeline/content.rs:397]: Yeah, that's probably a good idea. Thanks!

Comment (jmartinesp) [reply bindings/matrix-sdk-ffi/src/timeline/content.rs]: https://github.com/matrix-org/matrix-rust-sdk/pull/3801/commits/665c2c2f15dce87dd410bd250ca0a5bc81188503

File: crates/matrix-sdk-ui/src/timeline/event_item/content/pinned_events.rs [hunk 2]
```diff
@@ -0,0 +1,178 @@
+use std::collections::HashSet;
```
Comment (andybalaam) [inline crates/matrix-sdk-ui/src/timeline/event_item/content/pinned_events.rs:1]: Missing copyright.

**Review by andybalaam — APPROVED** [review]
Looks great, thank you! Just need to add a copyright notice to the new file.

File: crates/matrix-sdk-ui/src/timeline/event_item/content/pinned_events.rs (hunk 2 above)
Comment (jmartinesp) [reply crates/matrix-sdk-ui/src/timeline/event_item/content/pinned_events.rs:1]: Oh, weird. I expected the IDE to automatically add it 🤔 .

**Pull request merged**